spinoso-string/src/enc/ascii/mod.rs

```diff
@@ -343,6 +343,26 @@ impl AsciiString {
     }
 }
 
+// Index
+impl AsciiString {
+    #[inline]
+    #[must_use]
+    pub fn index(&self, needle: &[u8], offset: usize) -> Option<usize> {
+        let buf = self.get(offset..)?;
+        let index = buf.find(needle)?;
+        Some(index + offset)
+    }
+
+    #[inline]
+    #[must_use]
+    pub fn rindex(&self, needle: &[u8], offset: usize) -> Option<usize> {
+        let search_until = (offset + 1).min(self.len());
```

Comment:
we should use a saturating add on `offset` here to avoid an overflow.

Reply:
I added some tests in https://github.com/artichoke/artichoke/pull/2554 and fixed this.